services/cloudapi/cloudapinic/nic.go

```diff
@@ -0,0 +1,270 @@
+	Client *ionoscloud.APIClient
+	Meta   interface{}
+	D      *schema.ResourceData
+}
+
+func (fs *Service) Get(ctx context.Context, datacenterID, serverID string, depth int32) ([]ionoscloud.Nic, error) {
+	nics, apiResponse, err := fs.Client.NetworkInterfacesApi.DatacentersServersNicsGet(ctx, datacenterID, serverID).Depth(depth).Execute()
+	apiResponse.LogInfo()
+	if err != nil {
+		return nil, err
+	}
+	if apiResponse.HttpNotFound() || nics.Items == nil || len(*nics.Items) == 0 {
+		log.Printf("[DEBUG] no nics found for datacenter %s, server %s", datacenterID, serverID)
+		return nil, nil
+	}
+	return *nics.Items, nil
+}
+
+func (fs *Service) FindById(ctx context.Context, datacenterId, serverId, ID string, depth int32) (*ionoscloud.Nic, *ionoscloud.APIResponse, error) {
+	nic, apiResponse, err := fs.Client.NetworkInterfacesApi.DatacentersServersNicsFindById(ctx, datacenterId, serverId, ID).Depth(depth).Execute()
+	apiResponse.LogInfo()
+	if err != nil {
+		if apiResponse.HttpNotFound() {
+			log.Printf("[DEBUG] no nic found for datacenter %s, server %s", datacenterId, serverId)
+		}
+		return nil, apiResponse, err
+	}
+	return &nic, apiResponse, nil
+}
+
+func (fs *Service) Delete(ctx context.Context, datacenterID, serverID, ID string) (*ionoscloud.APIResponse, error) {
+	apiResponse, err := fs.Client.NetworkInterfacesApi.DatacentersServersNicsDelete(ctx, datacenterID, serverID, ID).Execute()
+	apiResponse.LogInfo()
+	if err != nil {
+		return apiResponse, err
+	}
+	// Wait, catching any errors
+	_, errState := cloudapi.GetStateChangeConf(fs.Meta, fs.D, apiResponse.Header.Get("Location"), schema.TimeoutDelete).WaitForStateContext(ctx)
+	if errState != nil {
+		return apiResponse, fmt.Errorf("an error occured while waiting for state change dcId: %s, server_id: %s, ID: %s, Response: (%w)", datacenterID, serverID, ID, errState)
+	}
+	return apiResponse, nil
+}
+
+// Create - creates the resource in the backend and waits until it is in an `AVAILABLE` state
+func (fs *Service) Create(ctx context.Context, datacenterID, serverID string, nic ionoscloud.Nic) (*ionoscloud.Nic, *ionoscloud.APIResponse, error) {
+	val, apiResponse, err := fs.Client.NetworkInterfacesApi.DatacentersServersNicsPost(ctx, datacenterID, serverID).Nic(nic).Execute()
+	apiResponse.LogInfo()
+	if err != nil {
+		return nil, apiResponse, fmt.Errorf("an error occured while creating val rule for dcId: %s, server_id: %s, Response: (%w)", datacenterID, serverID, err)
+	}
+	// Wait, catching any errors
+	_, errState := cloudapi.GetStateChangeConf(fs.Meta, fs.D, apiResponse.Header.Get("Location"), schema.TimeoutCreate).WaitForStateContext(ctx)
+	if errState != nil {
+		if cloudapi.IsRequestFailed(err) {
+			// Request failed, so resource was not created, delete resource from state file
+			fs.D.SetId("")
+		}
+		return nil, apiResponse, fmt.Errorf("an error occured while waiting for state change dcId: %s, server_id: %s, Response: (%w)", datacenterID, serverID, errState)
```

Comment:
@cristiGuranIonos same thing here, please include the `nic` word in the error message.